joomla/joomla-cms
# [5.3] Update tag router to restrict lossy matches to menu items set to any

Pull Request for Issue #40750 .

### Summary of Changes
Only populate the menu items into $this->lookup if they are a direct match, or return_any_or_all is set to any.


### Testing Instructions
Setup two tags.
Setup a menu item point to a singular tag
Setup a second menu item pointing to both tags, match type All
Populate articles into the first tag to show pagination within page (can change global config list limit to 5 to make this easier)
Navigate to the first menu item on the front end.
Observe pagination links

### Actual result BEFORE applying this Pull Request
Pagination URL incorrectly pointing to second menu item (with multiple tags selected), instead of the current menu item.
Page: /tag-1     pagination link:    /tag-1-and-2?start=5

Example: Page: /tag-1     pagination link:    /tag-1?start=5

### Expected result AFTER applying this Pull Request
Pagination URL's now correctly paginate through the current menu item.

Example: Page: /tag-1     pagination link:    /tag-1-and-2?start=5


### Link to documentations
Please select:
